### PR TITLE
Blog post on dnstap

### PR DESCRIPTION
**Pull request by varyoo**
Is the text clear enough?

### CODE REVIEW

**Comment by miekg** [comment]
Looks good! Spotted some small  typos. Maybe expand a little on where you
can send the dnstap data? Do you safe that or gzip?

File: content/blog/dnstap.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
++++
+title = "Logging with dnstap"
+description = "dnstap is a flexible, structured binary log format for DNS software."
+tags = ["dnstap", "log", "middleware"]
+draft = true
+date = "2017-08-01T16:25:28+02:00"
+author = "varyoo"
```
Comment (johnbelamaric) [inline content/blog/dnstap.md]: You need to add yourself to the defined list of authors in `config.toml`

File: content/blog/dnstap.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
++++
+
+*dnstap* is a flexible, structured binary log format for DNS software.
+It uses [Protocol Buffers](https://developers.google.com/protocol-buffers/) to encode events that occur inside DNS software in an implementation-neutral format.
+
+*dnstap* can log any DNS message exchanged by the server, along with information about the remote computer (IP address, port) and time.
+It includes client queries and responses, but also proxied requests or information requested from other name servers.
+
+A [*dnstap* middleware] has been added in [CoreDNS-010]({{< relref "blog/coredns-010.md" >}}).
+Currently it can only log client level messages. Logging additional type of exchanges is being experimented.
```
Comment (johnbelamaric) [inline content/blog/dnstap.md]: s/being experimented/under development/

File: content/blog/dnstap.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    
+    ;; QUESTION SECTION:
+    ;example.org.       IN       A
+
+    ;; ANSWER SECTION:
+    example.org.        86339   IN      A       93.184.216.34
+~~~
+
+The [*dnstap* middleware] is used in combinaison with the *dnstap* command-line tool.
+The latest listen the same socket the other is logging to.
```
Comment (johnbelamaric) [inline content/blog/dnstap.md]: s/latest listen/CLI tool listens/
s/other/middleware/

File: content/blog/dnstap.md [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    ;; flags: qr rd ra; QUERY: 1, ANSWER: 1, AUTHORITY: 0, ADDITIONAL: 0
+    
+    ;; QUESTION SECTION:
+    ;example.org.       IN       A
+
+    ;; ANSWER SECTION:
+    example.org.        86339   IN      A       93.184.216.34
+~~~
+
+The [*dnstap* middleware] is used in combinaison with the *dnstap* command-line tool.
```
Comment (johnbelamaric) [inline content/blog/dnstap.md]: s/combinaison/combination/

File: config.toml [hunk 5]
```diff
@@ -37,6 +37,10 @@ theme = "coredns"
   name = "Jonathan Dickinson"
   github = "https://github.com/jcdickinson"
 
+[params.authors.varyoo]
+  name = "Antoine D."
+  github = "https://github.com/varyoo"
```
Comment (miekg) [inline config.toml]: do you want a @coredns.io email address?

File: content/blog/dnstap.md [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+    
+    ;; QUESTION SECTION:
+    ;example.org.       IN       A
+
+    ;; ANSWER SECTION:
+    example.org.        86339   IN      A       93.184.216.34
+~~~
+
+A [*dnstap* middleware] has been added in [CoreDNS-010]({{< relref "blog/coredns-010.md" >}}).
+Currently it can only log client level messages. Logging additional type of exchanges is being experimented.
```
Comment (miekg) [inline content/blog/dnstap.md]: -> being implemented

File: content/blog/dnstap.md [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+
+# Examples
+
+Add *dnstap* to the *Corefile*:
+
+~~~ text
+dnstap /tmp/dnstap.sock full
+~~~
+
+Listen on the *dnstap* socket and write message payloads to *stdout*:
```
Comment (miekg) [inline content/blog/dnstap.md]: Now you can use the `dnstap` tool (link to their site) to read from the socket where CoreDNS write to. With the *full* option given to the `dnstap` middleware you will also include the full (binary) data of the DNS message.

File: config.toml (hunk 5 above)
Comment (varyoo) [inline config.toml]: I could add varyoo@coredns.io here, thanks!

File: content/blog/dnstap.md (hunk 7 above)
Comment (varyoo) [inline content/blog/dnstap.md]: It is not clear how to find the CLI tool, even on their website, I added a footnote link.

**Comment by miekg** [comment]
[ Quoting <notifications@github.com> in "Re: [coredns/coredns.io] Blog post ..." ]
>varyoo commented on this pull request.
>
>
>
>> @@ -37,6 +37,10 @@ theme = "coredns"
>   name = "Jonathan Dickinson"
>   github = "https://github.com/jcdickinson"
>
>+[params.authors.varyoo]
>+  name = "Antoine D."
>+  github = "https://github.com/varyoo"
>
>I could add varyoo@coredns.io here, thanks!

Added, it forwards to your gmail address.

**Comment by miekg** [comment]
Will push this out tomorrow! Thanks for writing.

**Pull request merged**